src/docfx/lib/github/GitHubUserCache.cs

```diff
@@ -126,20 +127,37 @@ public async Task<(Error, GitHubUser)> GetByCommit(string authorEmail, string re
             return await GetByLogin(login);
         }
 
-        public Task SaveChanges()
+        public async Task<Error> SaveChanges(Config config)
         {
             if (!_updated)
             {
-                return Task.CompletedTask;
+                return null;
             }
 
+            string file;
             lock (_lock)
             {
-                var file = new GitHubUserCacheFile
+                file = JsonUtility.Serialize(new GitHubUserCacheFile
                 {
                     Users = Users.ToArray(),
-                };
-                return ProcessUtility.WriteFile(_cachePath, JsonUtility.Serialize(file));
+                });
+            }
+            await ProcessUtility.WriteFile(_cachePath, file);
+            var url = config.GitHub.UserCache;
+            if (!config.GitHub.UpdateRemoteUserCache || !HrefUtility.IsHttpHref(url))
+            {
+                return null;
+            }
+
+            try
+            {
+                // TOOD: aware of ETag to handle conflicts
+                await HttpClientUtility.PutAsync(config.GitHub.UserCache, new StringContent(file), config);
```

Comment:
Shall we retry within `PutAsync` as we did in `GetAsync`? Or these two could be consolidated.

Reply:
Want to on hold this change as the retry may include checking etag and merge cache.